util/travis_utils.sh

```diff
@@ -12,6 +12,7 @@ TRAVIS_COMMIT_RANGE="${TRAVIS_COMMIT_RANGE:-HEAD~1..HEAD}"
 # Extra variables
 LOCAL_BRANCH=$(git rev-parse --abbrev-ref HEAD)
 QMK_CHANGES=$(git diff --name-only -n 1 ${TRAVIS_COMMIT_RANGE})
+NUM_CORE_CHANGES=$(echo "$QMK_CHANGES" | grep -Ec -e '^Makefile' -e '^[^/]*.mk' -e '^drivers/' -e '^lib/atsam' -e '^lib/lib8tion/' -e '^platforms' -e '^quantum' -e '^tests' -e '^tmk_core')
```

Comment:
If I have understood the grep correctly, are the submodules excluded here for a reason?

I can see a valid cases both ways:
* we want to build everything when a submodule changes to check for compatibility
* users unknowingly push old/wrong submodules

Reply:
I'm happy to go either way here. I don't have a reason to have excluded them other than initially pulling in things that are actually in qmk_firmware.

For #2 we might be better off having a lint check anyway.